include/deal.II/base/index_set.h

```diff
@@ -261,7 +261,7 @@ class IndexSet
    * n_elements().
    */
   size_type
-  nth_index_in_set(const unsigned int local_index) const;
+  nth_index_in_set(const size_type local_index) const;
```

Comment:
I am not quite sure about this one. If we want to keep using `unsigned int` for local sizes that we don't expect to be large, I would rather keep it. If we just want to be safe, this makes sense but might not be in line with other places.

Reply:
Since `IndexSet` stores ranges (rather than individual entries like a `std::set`), there is nothing wrong with having an index set of size 10B and asking for the 5B'th entry.

Reply:
I agree with Wolfgang, especially because it doesn't cost much (if you store local indices as ints somewhere else, they will just be converted when passed).